rustsat/src/types.rs

```diff
@@ -697,6 +704,59 @@ impl Assignment {
             Some(var![self.assignment.len() as u32 - 1])
         }
     }
+
+    //Read a solution from a SAT solver's output
+    pub fn from_solver_output_path<P: AsRef<Path>>(path: P) -> anyhow::Result<SolverOutput> {
+        let reader = std::io::BufReader::new(
+            instances::fio::open_compressed_uncompressed_read(path)
+                .context("failed to open reader")?,
+        );
+        instances::fio::parse_sat_solver_output(reader)
+    }
+
+    pub fn from_vline(line: &str) -> anyhow::Result<Self> {
+        if !line.starts_with('v') {
+            panic!("Value line does not start with v.");
+        }
+
+        let line = &line[2..];
+        let mut assignment = Assignment::default();
+        for number in line.trim().split(' ').filter(|x| !x.is_empty()) {
+            let number_v = number.parse::<i32>().unwrap();
+
+            //End of the value lines
+            if number_v == 0 {
+                continue;
+            }
+
+            let literal = Lit::from_ipasir(number_v)?;
+            assignment.assign_lit(literal);
+        }
+
+        Ok(assignment)
+    }
+
+    /// Parses and saves literals from value line.
+    pub fn extend_from_vline(&mut self, line: &str) -> anyhow::Result<()> {
+        if !line.starts_with('v') {
+            panic!("Value line does not start with v.");
```

Comment:
Sorry in case the `#[should_panic]` on the tests was confusing. As a rule of thumb, never use `panic` inside the library, but always return errors. Once your error types are fixed, we can update the tests to expect the exact error rather than simply panic.
For here, please introduce a `InvalidVLine` enum with at least the variants `InvalidTag(char)` and `ConflictingAssignment(Var)` and use them appropriately.